.github/workflows/flatpak.yml

```diff
@@ -21,7 +21,7 @@ jobs:
   publish:
     name: Publish to Flathub
     runs-on: [ubuntu-latest]
-    if: "${{ github.event_name == 'release' && ( matrix.branch != 'stable' || (!contains(github.ref, '-beta') && !contains(github.ref, '-rc')) ) }}"
```

Comment:
It's annoying that the condition can never work that way, but I feel the construct itself isn't great.

So what are the conditions for the job to run? I feel the logic (triggering two jobs in a matrix) which then internally check whether they should run at all is wasteful.

From what I could tell the flow is this:

* There needs to have been an OBS release
* On _any_ version number, a beta release will be uploaded to FlatHub
* On _any_ version number without the `-beta` or `-rc` suffix, a stable release will be uploaded

If I'm reading the logic right, a version tag `27.5.2-YOLO` would lead to a stable release being uploaded, which I guess is not intended.

Instead, I'd do the following:

* There needs to have been an OBS release
* Run a single shell script step that does the following:
  * Check the ref for suffixes
  * If the ref has suffixes, assume non-stable release
  * Set output variables for flat-manager-url, repository, and token based on whether stable or beta is required
* Use the output variables in a single invocation of `flat-manager` with the output variables set before